db/migrations/20210614142629_split_out_job_links.js

```diff
@@ -0,0 +1,88 @@
+
+exports.up = function(knex) {
+  // Modify the jobs table to add a jobID
+  return knex.schema
+  .alterTable('jobs', (t) => {
+    t.uuid('jobID');
+    t.unique('jobID');
+  })
+  .then(() => {
+    // Populate the jobID to match the requestID
+    return knex('jobs').update({
+      jobID: knex.raw('??', ['requestId'])
+    });
+  })
+  .then(() => {
+    // Create the job links table
+    return knex.schema.createTable('job_links', (t) => {
+      t.increments('id')
+        .primary();
+
+      t.uuid('jobID')
+        .notNullable()
+        .references('jobID')
+        .inTable('jobs')
+        .onDelete('CASCADE');
+
+      t.string('href', 4096)
+        .notNullable();
+
+      t.string('type', 255);
+
+      t.string('title', 255);
+
+      t.string('rel', 255);
+
+      t.timestamp('temporalStart');
+
+      t.timestamp('temporalEnd');
+
+      t.string('bbox', 255);
+
+      t.timestamp('createdAt')
+        .notNullable();
+
+      t.timestamp('updatedAt')
+        .notNullable();
+
+      t.index(['jobID']);
+    });
+  })
+  .then(async () => {
+    // Populate the job_links table
+    const jobs = await knex.select(['jobID', '_json_links']).from('jobs').forUpdate();
+    jobs.forEach(async (job) => {
+      let links = job._json_links;
+      if (typeof links === 'string') {
+        links = JSON.parse(links);
+      }
+      if (links && links.length > 0) {
```

Comment:
Will definitely want to watch this as it deploys to SIT and UAT 😨

Reply:
Tested it in Sandbox - it looked fine for the 18 jobs there. I still have a bunch of worries - potentially taking awhile with many jobs, memory issues (I think forEach should just process one thing in memory at a time, but we're going to be loading all the jobs from the database at once), not loading all the jobs for some reason, unexpected data, etc. Will definitely need to watch.